### DISCUSSION TITLE
Custom Field Import - Type Errors

### DISCUSSION THREAD

**Discussion by adfrost81** [netbox-community/netbox#9616] [Q&A] [answered]
This is probably something super simple that I am missing, but I'm not finding adequate guidance [in the documentation](https://docs.netbox.dev/en/stable/customization/custom-fields/). I'm trying to import a bunch of custom fields I've defined. 16 of them are of type Selection and Multiple Selection. However, when I try to do the import, I keep getting these kind of errors:

"Row 1 type: Select a valid choice. selection is not one of the available choices."
"Row 1 choices: Choices may be set only for custom selection fields."

I've tried formatting the type in all lower case and with capitals, but it doesn't seem to matter. I did an export to see how it's indicated there, and it exports with "Selection" capitalized.

Here's a snippet of the .csv data:

`name,label,type,description,content_types,weight,filter_logic,choices`
`st_onsite,On Site Support,selection,Team assigned for on-site support.,dcim.site,106,loose,"CHS, Dakota Panel, Holiday City DC, Iron Ridge, Plano DC, Terre Haute DC, Valley DC"`
`st_color_printers,Color LaserJet Model,multiple selection,Indicates what model color laser printers are installed in the departments.,dcim.site,210,loose,"HP LaserJet CP2025n, HP LaserJet M451dn, HP LaserJet M452dn, HP LaserJet M454dn"`

**Comment by julianstolp** [comment, marked as answer]
You almost got it right. Just change the typevalues to select and multiselect as stated in the hint ![image](https://user-images.githubusercontent.com/41152397/176843750-51aa4cfe-f5bf-4611-b19a-77f4ae48df4c.png) at the CSV Field Options

**Comment by adfrost81** [comment]
Thanks! That worked. For some reason, I don't have the hint icon by the Type field, otherwise I probably wouldn't have needed to ask this question. I couldn't find it documented anywhere else.

![FieldOptions](https://user-images.githubusercontent.com/91818709/176974603-bb69e645-eeaf-452a-8ddc-57133aa9d572.png)

**Reply by candlerb** [reply]
Which version of Netbox are you using?

With v3.2.5 I see:
![image](https://user-images.githubusercontent.com/44789/176990721-18c31e21-eed4-4c48-9af0-bcfae3a3de61.png)

The help text "Field data type (e.g. text, integer, etc.)" was added in December 2021:

```
commit 43fadab3bbb770d4914e159690cbcc66f24a890f
Author: jeremystretch <jstretch@ns1.com>
Date:   Wed Dec 29 11:57:27 2021 -0500

    Closes #8034: Enable specifying custom field validators during CSV import
```

and the `object_type` field was added recently:

```
commit 998a392bd3f67ee13c01cb099a189cc09c797385
Author: jeremystretch <jstretch@ns1.com>
Date:   Tue May 31 11:37:30 2022 -0400

    Fixes #9425: Fix bulk import for object and multi-object custom fields
```

Hence you'll probably find it worthwhile upgrading.

**Reply by adfrost81** [reply]
That explains it. The instance I'm trying to import this data on is v 3.1.0. This is just an eval instance and the one we will be actually implementing for production is on 3.2.3, apparently because some plugins aren't working on the latest version yet. Those tool tips are a lot more handy.
